src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,509 @@
+    }
+
+    /**
+     * Returns a {@link RandomGenerator} that utilizes the {@code name} algorithm.
+     *
+     * @param name      Name of random number algorithm to use
+     * @param category  Sub-interface of {@link RandomGenerator} to type check
+     * @param <T>       Sub-interface of {@link RandomGenerator} to produce
+     *
+     * @return An instance of {@link RandomGenerator}
+     *
+     * @throws IllegalArgumentException when either the name or category is null
+     */
+    static <T> T of(String name, Class<? extends RandomGenerator> category)
+            throws IllegalArgumentException {
+        @SuppressWarnings("unchecked")
+        T uncheckedRandomGenerator = (T)findProvider(name, category).get();
+        return uncheckedRandomGenerator;
+    }
+
+    /**
+     * Returns a {@link RandomGeneratorFactory} that will produce instances
+     * of {@link RandomGenerator} that utilizes the {@code name} algorithm.
+     *
+     * @param name  Name of random number algorithm to use
+     * @param category Sub-interface of {@link RandomGenerator} to type check
+     * @param <T> Sub-interface of {@link RandomGenerator} to produce
+     *
+     * @return Factory of {@link RandomGenerator}
+     *
+     * @throws IllegalArgumentException when either the name or category is null
+     */
+    static <T extends RandomGenerator> RandomGeneratorFactory<T> factoryOf(String name, Class<? extends RandomGenerator> category)
+            throws IllegalArgumentException {
+        Provider<? extends RandomGenerator> uncheckedProvider = findProvider(name, category);
+        return new RandomGeneratorFactory<>(uncheckedProvider);
+    }
+
+    /**
+     * Fetch the required constructors for class of random number algorithm.
+     *
+     * @param randomGeneratorClass class of random number algorithm (provider)
+     */
+    @SuppressWarnings("unchecked")
+    private synchronized void getConstructors(Class<? extends RandomGenerator> randomGeneratorClass) {
+        if (ctor == null) {
+            PrivilegedExceptionAction<Constructor<?>[]> ctorAction = randomGeneratorClass::getConstructors;
+            try {
+                Constructor<?>[] ctors = AccessController.doPrivileged(ctorAction);
+                for (Constructor<?> ctorGeneric : ctors) {
+                    Constructor<T> ctorSpecific = (Constructor<T>)ctorGeneric;
+                    final Class<?>[] parameterTypes = ctorSpecific.getParameterTypes();
+
+                    if (parameterTypes.length == 0) {
+                        ctor = ctorSpecific;
+                    } else if (parameterTypes.length == 1) {
+                        Class<?> argType = parameterTypes[0];
+
+                        if (argType == long.class) {
+                            ctorLong = ctorSpecific;
+                        } else if (argType == byte[].class) {
+                            ctorBytes = ctorSpecific;
+                        }
+                    }
+                }
```

Comment:
Add an `assert` statement that `ctor`, `ctorLong` and `ctorBytes` are all non-null?

Reply:
Only `ctor` is required but yes.